src/Psl/Fun/Internal/LazyEvaluator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Fun\Internal;
+
+/**
+ * @internal
```

Comment:
Not sure this should be internal, since psalm complains about typehinting the results with `callable(): ExpectedResult`

Reply:
(Fixed in the `lazy()` method by returning a closure around the evaluator)